src/Microsoft.OData.Client/ALinq/DataServiceQueryProvider.cs

```diff
@@ -152,5 +176,82 @@ internal QueryComponents Translate(Expression e)
             LambdaExpression selector = re.Projection == null ? null : re.Projection.Selector;
             return new QueryComponents(uri, version, lastSegmentType, selector, normalizerRewrites);
         }
+
+        /// <summary>
+        /// Parses the result of a query set count request.
+        /// </summary>
+        /// <typeparam name="TElement">The return type.</typeparam>
+        /// <param name="queryResult">The query result.</param>
+        /// <returns></returns>
+        private static TElement ParseQuerySetCount<TElement>(QueryResult queryResult)
+        {
+            StreamReader reader = new StreamReader(queryResult.GetResponseStream());
+            long querySetCount = -1;
+
+            try
+            {
+                querySetCount = XmlConvert.ToInt64(reader.ReadToEnd());
+            }
+            finally
+            {
+                reader.Close();
+            }
+
+            return (TElement)Convert.ChangeType(querySetCount, typeof(TElement), System.Globalization.CultureInfo.InvariantCulture.NumberFormat);
+        }
+
+        /// <summary>
+        /// Parses the scalar result of an aggegrate request.
+        /// </summary>
+        /// <typeparam name="TElement">The return type.</typeparam>
+        /// <param name="queryResult">The query result.</param>
+        /// <returns></returns>
+        private TElement ParseAggregateSingletonResult<TElement>(QueryResult queryResult)
+        {
+            IDictionary<string, string> responseHeaders = new Dictionary<string, string>();
+            responseHeaders.Add(ODataConstants.ContentTypeHeader, "application/json");
+            HttpWebResponseMessage httpWebResponseMessage = new HttpWebResponseMessage(responseHeaders, (int)queryResult.StatusCode, queryResult.GetResponseStream);
+
+            ODataMessageReaderSettings messageReaderSettings = new ODataMessageReaderSettings
+            {
+                Validations = ~ValidationKinds.ThrowOnUndeclaredPropertyForNonOpenType

```

Comment:
What does the `~` operator do in this case?

Reply:
@habbes The ~ operator performs a bitwise complement operation on its operand. In this case, we're configuring the message reader NOT to throw on undeclared property for non-open type.